src/Transform/BundleMemoryPools.cpp

```diff
@@ -34,7 +34,8 @@ namespace {
 //===----------------------------------------------------------------------===//
 
 // Data structure for managing memory pools.
-typedef std::map<Block *, AllocOp> BlockToMemPool;
+typedef std::map<int64_t, AllocOp> AlignmentToMemPool;
```

Comment:
Do you mind adding a little more explanation on these two maps? Describing data goes a long way toward making the code more readable. My understanding is that each block has a unique map of alignment to a mem pool". So we have one alloc pool per alignment and block.

It would be great if you could also write at the top of this (or the header file) what are the high level characteristics currently handled by your approach:

Also currently bundle data that satisfy x) y) z) Currently bundling of buffers that w) is prevented.
This would help us better understand the code as it evolves.

Reply:
The changes are minimal, instead of having a single memory pool as before now there is one per alignment.